JamJar/JamJarPageViewController.swift

```diff
@@ -0,0 +1,371 @@
+                    self.video.userVote = false
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        }
+    }
+    
+    @IBAction func flagVideoTapped(sender: UIButton) {
+        let videoId = self.video.id!
+        
+        let flagView = SCLAlertView()
+        
+        // Creat the subview
+        let subview = UIView(frame: CGRectMake(0,0,216,100))
+        let x = (subview.frame.width - 180) / 2
+        
+        // reason label
+        let reasonLabel = UILabel(frame: CGRectMake(x,5,180,25))
+        reasonLabel.text = "Reason:"
+        subview.addSubview(reasonLabel)
+        
+        // reason PickerView
+        let reasonPickerView = UIPickerView(frame: CGRectMake(x,10,180,100))
+        reasonPickerView.delegate = self
+        reasonPickerView.dataSource = self
+        subview.addSubview(reasonPickerView)
+        
+        // Add the subview to the alert's UI property
+        flagView.customSubview = subview
+        
+        // Add Notes Text Field
+        let notes = flagView.addTextField("Notes")
+        
+        flagView.addButton("Submit") {
+            let reason = self.reasonsToFlag[reasonPickerView.selectedRowInComponent(0)]
+            
+            var flagType: String
+            if reason == "Report User" {
+                flagType = "U"
+            } else {
+                flagType = String(reason[reason.startIndex])
+            }
+            
+            VideoService.flag(videoId, reason: flagType, notes: notes.text) { success, result in
+                if !success {
+                    // Error - show the error
+                    let errorTitle = "Video Flag Error!"
+                    SCLAlertView().showError(errorTitle, subTitle: result!, closeButtonTitle: "Got it")
+                } else {
+                    // Success
+                    let successTitle = "Video Flagged!"
+                    SCLAlertView().showSuccess(successTitle, subTitle: result!, closeButtonTitle: "Got it")
+                }
+            }
+        }
+        
+        flagView.showEdit("Flag Video", subTitle: "Reason", closeButtonTitle: "Cancel")
+    }
+    
+    //Allow rotate
+    override func shouldAutorotate() -> Bool {
+        return true
+    }
+    
+    override func supportedInterfaceOrientations() -> UIInterfaceOrientationMask {
+        return UIInterfaceOrientationMask.All
+    }
+    
+    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
+        switch UIDevice.currentDevice().orientation{
+        case .Portrait:
+            unfullScreenVideo()
+            break
+        default:
+            fullScreenVideo()
+        }
+    }
+    
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        if(segue.identifier == "playJamJar") {
+            unowned let embeddedVideoViewController = segue.destinationViewController as! StitchedJamJarAVPlayerViewController
```

Comment:
We should add some protection here to check if the hls_src exists here so the app doesn't crash if the video is still processing